ipn/ipnserver/server.go

```diff
@@ -238,12 +238,26 @@ func bufferHasHTTPRequest(br *bufio.Reader) bool {
 		mem.Contains(mem.B(peek), mem.S(" HTTP/"))
 }
 
+// bufferIsConnect reports whether br looks like it's likely an HTTP
+// CONNECT request.
+func bufferIsConnect(br *bufio.Reader) bool {
+	peek, _ := br.Peek(br.Buffered())
```

Comment:
Why Buffered instead of 4 (or `len("CONN")`)? As is it works b/c this gets called from serveConn that does a peek(4) already, but changing that might then break this...

Reply:
I added an "Invariant: ..." doc instead to the func's docs.

I don't want the Peek to cause any actual I/O to happen. I want to only use what's available. So I instead documented the caller must've already peeked at least 4 bytes.